src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Unix.targets

```diff
@@ -120,19 +137,10 @@ The .NET Foundation licenses this file to you under the MIT license.
       <LinkerArg Include="-Wl,--as-needed" Condition="'$(_IsApplePlatform)' != 'true'" />
       <LinkerArg Include="-Wl,-e0x0" Condition="'$(NativeLib)' == 'Shared' and '$(_IsApplePlatform)' != 'true'" />
       <LinkerArg Include="-pthread" Condition="'$(_IsApplePlatform)' != 'true'" />
-      <LinkerArg Include="-lstdc++" Condition="'$(LinkStandardCPlusPlusLibrary)' == 'true'" />
-      <LinkerArg Include="-ldl" />
-      <LinkerArg Include="-lobjc" Condition="'$(_IsApplePlatform)' == 'true'" />
-      <LinkerArg Include="-lswiftCore" Condition="'$(_targetOS)' == 'osx'" />
-      <LinkerArg Include="-lswiftFoundation" Condition="'$(_targetOS)' == 'osx'" />
-      <LinkerArg Include="-lz" />
-      <LinkerArg Include="-lrt" Condition="'$(_IsApplePlatform)' != 'true' and '$(_linuxLibcFlavor)' != 'bionic'" />
-      <LinkerArg Include="-llog" Condition="'$(_linuxLibcFlavor)' == 'bionic'" />
-      <LinkerArg Include="-licucore" Condition="'$(_IsApplePlatform)' == 'true'" />
+      <LinkerArg Include="@(NativeSystemLibrary->'-l%(Identity)')" />
       <LinkerArg Include="-L/usr/lib/swift" Condition="'$(_targetOS)' == 'osx'" />
       <LinkerArg Include="@(StaticICULibs)" Condition="'$(StaticICULinking)' == 'true'" />
       <LinkerArg Include="@(StaticSslLibs)" Condition="'$(StaticOpenSslLinking)' == 'true'" />
```

Comment:
We can move these three as well.

Reply:
The first one is capital L so it doesn't feel like it fits either of the new ItemGroups. Would need a `NativeSystemLibraryPath`

Breaking up the static ICU/SSL weirdness would be more diffs so it's the same reasoning as "We could try to put more arguments on the ExtraLinkerArg plan but I wanted to keep the diff short and reviewable for now."